encrypted-media/util/drm-messagehandler.js

```diff
@@ -0,0 +1,230 @@
+    },
+
+    "com.microsoft.playready": {
+        "serverURL": "https://lic.staging.drmtoday.com/license-proxy-headerauth/drmtoday/RightsManager.asmx",
+        "servertype" : "drmtoday",
+        "userId" : "12345",
+        "merchant" : "cablelabs"
+    }
+*/
+
+};
+
+function MessageHandler( keysystem, content, sessionType ) {
+    this._keysystem = keysystem;
+    this._content = content;
+    this._sessionType = sessionType || "temporary";
+    this._drmconfig = drmconfig[ this._keysystem ];
+
+    this.messagehandler = MessageHandler.prototype.messagehandler.bind( this );
+    if ( this._drmconfig && this._drmconfig.certificate ) {
+        this.servercertificate = stringToUint8Array( atob( this._drmconfig.certificate ) );
+    }
+}
+
+MessageHandler.prototype.messagehandler = function messagehandler( messageType, message) {
+
+    const keySystems = {
+        'com.widevine.alpha': {
+            responseType: 'json',
+            getLicenseMessage: function(response) {
+                return base64DecodeToUnit8Array( response.license );
+            },
+            getErrorResponse: function(response) {
+                return response;
+            },
+            getLicenseRequestFromMessage: function(message) {
+                return new Uint8Array(message);
+            },
+            getRequestHeadersFromMessage: function(/*message*/) {
+                return null;
+            }
+        },
+        'com.microsoft.playready': {
```

Comment:
Agreed. But this is a function of the server protocol. The protocol supported by the DRMToday servers expects the raw message contents for Widevine but expects some extracted part of the message for PlayReady. The code below that extracts part of the PlayReady message could be on the server side.